core/timeline_api.php

```diff
@@ -57,13 +57,25 @@ function timeline_events( $p_start_time, $p_end_time, $p_max_events, $p_filter =
 	$t_result = history_query_result( $t_query_options );
 	$t_count = 0;
 
+	$t_previous_history_event = null;
 	while ( $t_history_event = history_get_event_from_row( $t_result, /* $p_user_id */ auth_get_current_user_id(), /* $p_check_access_to_issue */ true ) ) {
 		$t_event = null;
-		$t_user_id = $t_history_event['userid'];
+		$t_user_id = (int)$t_history_event['userid'];
 		$t_timestamp = $t_history_event['date'];
 		$t_issue_id = $t_history_event['bug_id'];
+		$t_type = $t_history_event['type'];
 
-		switch( $t_history_event['type'] ) {
+		if ( $t_previous_history_event != null ) {  //check repeated event
```

Comment:
We don't combine similar actions, e.g. adding multiple tags.  I wonder if we should just keep it as an entry per file.

Reply:
I agree. And in fact this is introducing a regression, e.g. when adding multiple tags in a single operations: only one timeline entry with the last processed tag is shown, instead of one per tag before this change.